### ISSUE TITLE
[BUG] Type mismatched in Utils.ts

### ISSUE THREAD

**Issue by altro3** [javascript-obfuscator/javascript-obfuscator#971]
Can't compile project with javascipt obfuscator `2.18.1`. Got this error:

```
node_modules\javascript-obfuscator\src\utils\Utils.ts
24:52-66
[tsl] ERROR in node_modules\javascript-obfuscator\src\utils\Utils.ts(24,53)
      TS2345: Argument of type 'string | undefined' is not assignable to parameter of type 'string'.
  Type 'undefined' is not assignable to type 'string'.
ts-loader-default_e3b0c44298fc1c14
```
Problem block:
```
    public static buildVersionMessage (version?: string, buildTimestamp?: string): string {
        const isUnknownVersion = !version || !buildTimestamp;

        if (isUnknownVersion) {
            return 'unknown';
        }

        const buildDate: string = new Date(parseInt(buildTimestamp, 10)).toISOString();

        return `${version}_${buildDate}`;
    }
```

Need to add **!** after buildTimestamp:
```
const buildDate: string = new Date(parseInt(buildTimestamp!, 10)).toISOString();
```

**Comment by sanex3339** [comment]
Hi. Please update your dependencies. Current obfuscator version uses `4.4.0-beta`

**Comment by jakebailey** [comment]
This has affected something I'm updating, and I'm a bit surprised that the package includes its `src` directory in the package and depends on that to work; I haven't personally seen a package that has a handwritten `index.d.ts` file that exports form the sources, only from generated ones that `tsc` outputs for the purposes of packaging. Relying on the sources means that anyone who is depending on the types is also wasting some time type checking regular `.ts` files.

Users of packages that are written in TS shouldn't have to worry about the TS version that the dependency uses to compile; the `d.ts` files are supposed to be the intermediary (where the compiled JS output files are the actual runtime bits). I think you can prevent these sorts of issues by instead using `--declaration` (or the tsconfig), which will still allow you to write the `index.d.ts` file as it is now (though better would be to not need to have it at all, instead relying on the generated `d.ts` files).

**Comment by sanex3339** [comment]
Hmm. Good point.

**Comment by sanex3339** [comment]
I will reopen this until 2.19.0. I'll try to rework publish process

**Comment by sanex3339** [comment]
https://github.com/javascript-obfuscator/javascript-obfuscator/pull/972/commits/0306f29ba87f176632dba4c42347e1f398435bec
https://github.com/javascript-obfuscator/javascript-obfuscator/pull/972/commits/26acaec12a60937a9e9512f86ec6303fb46694ab

Here i reworked how publish and commit processes are working

**Comment by jakebailey** [comment]
Thanks! I'll be happy to test it out once it's released.

**Comment by sanex3339** [comment]
2.19.0 version was released. Please check it.

**Comment by sanex3339** [comment]
Should be fixed in `2.19.0`. Please let me know if it's not.

**Comment by jakebailey** [comment]
Looks good to me, thanks.
